execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1722,34 +1722,16 @@ where
         Ok(self.state.get_trie_full(correlation_id, &trie_key)?)
     }
 
-    /// Puts a trie and finds missing descendant trie keys.
-    pub fn put_trie_and_find_missing_descendant_trie_keys(
-        &self,
-        correlation_id: CorrelationId,
-        trie_bytes: &[u8],
-    ) -> Result<Vec<Digest>, Error>
+    /// Puts a trie.
+    // TODO: Offer to verify the trie has all of its descendents present?
+    // TODO: Do we want to return the calculated inserted trie key?
+    pub fn put_trie(&self, correlation_id: CorrelationId, trie_bytes: &[u8]) -> Result<(), Error>
     where
         Error: From<S::Error>,
     {
-        let inserted_trie_key = self.state.put_trie(correlation_id, trie_bytes)?;
-        let missing_descendant_trie_keys = self
-            .state
-            .missing_trie_keys(correlation_id, vec![inserted_trie_key])?;
-        Ok(missing_descendant_trie_keys)
-    }
+        let _inserted_trie_key = self.state.put_trie(correlation_id, trie_bytes)?;
```

Comment:
Should we check whether all children (not descendants!) are present, and return an error if not?

On the other hand that would make it slower, of course. We could offer a `_no_check` version of the method if the caller is confident all children were stored first.

In either case the method documentation should mention the new invariant.